src/handlers/expiry/providers/TestDataProvider.ts

```diff
@@ -18,13 +18,12 @@ export class TestDataProvider implements ITestDataProvider {
     filters: models.ITestResultFilters
   ): Promise<models.ITestResult[]> {
     try {
-      this.testResultsDAO = this.testResultsDAO as models.TestResultsDAO;
-      const result = await this.testResultsDAO.getBySystemNumber(
+      const result = await this.testResultsDAO?.getBySystemNumber(
```

Comment:
`testResultsDAO` can be null here? I think we can enforce to have a valid DAO here

Reply:
The current implementation always initialises testResultDAO so testResultsDAO is not null. However, I have made some changes to TestDataProvider which are only possible in this branch as all code is refactored but they impact the unit tests. I will discuss this with you in detail.